lib/modules/manager/crow/extract.ts

```diff
@@ -0,0 +1,64 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { parseSingleYaml } from '../../../util/yaml';
+import { getDep } from '../dockerfile/extract';
+import type { ExtractConfig, PackageFileContent } from '../types';
+import type { CrowConfig } from './types';
+
+function crowVersionDecider(CrowConfig: CrowConfig): (keyof CrowConfig)[] {
+  const keys = ['clone', 'steps', 'pipeline', 'services'];
+  return Object.keys(CrowConfig).filter((key) =>
+    keys.includes(key),
+  ) as (keyof CrowConfig)[];
+}
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string,
+  extractConfig: ExtractConfig,
+): PackageFileContent | null {
+  logger.debug('crow  .extractPackageFile()');
+  let config: CrowConfig;
+  try {
+    // TODO: use schema (#9610)
+    config = parseSingleYaml(content);
```

Comment:
can you create a zod schema please? so this manager is an improvement and not just a copy?